src/main/java/com/google/firebase/database/connection/NettyWebSocketClient.java

```diff
@@ -57,38 +60,51 @@
   private final ChannelHandler channelHandler;
   private final ExecutorService executorService;
   private final EventLoopGroup group;
+  private final boolean isSecure;
 
   private Channel channel;
 
   NettyWebSocketClient(
       URI uri, String userAgent, ThreadFactory threadFactory,
-      WebsocketConnection.WSClientEventHandler eventHandler) {
+      WebsocketConnection.WSClientEventHandler eventHandler, boolean isSecure) {
     this.uri = checkNotNull(uri, "uri must not be null");
     this.eventHandler = checkNotNull(eventHandler, "event handler must not be null");
     this.channelHandler = new WebSocketClientHandler(uri, userAgent, eventHandler);
     this.executorService = new FirebaseScheduledExecutor(threadFactory,
         "firebase-websocket-worker");
     this.group = new NioEventLoopGroup(1, this.executorService);
+    this.isSecure = isSecure;
   }
 
   @Override
   public void connect() {
     checkState(channel == null, "channel already initialized");
+    SslContext sslContext = null;
+    if (this.isSecure) {
+      TrustManagerFactory trustFactory = null;
+      try {
+        trustFactory = TrustManagerFactory.getInstance(
+            TrustManagerFactory.getDefaultAlgorithm());
+        trustFactory.init((KeyStore) null);
+        sslContext = SslContextBuilder.forClient()
+            .trustManager(trustFactory).build();
+      } catch (NoSuchAlgorithmException | KeyStoreException | SSLException e) {
+        eventHandler.onError(e.getCause());
```

Comment:
Can you change this back to the original structure (one big try/catch block surrounding everything)? Right now, if the SSL initialization fails, the code below still executes.

Reply:
Done